### PR TITLE
feat: add h3 to benchmarks

### PR DESCRIPTION
**Pull request by manniL**
Updated version of #164 ☺️ 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: benchmarks/h3.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+'use strict'
+
+const { createServer } = require('http')
+const { createApp, toNodeListener, eventHandler } = require('h3')
+
+const app = createApp()
+app.use('/', eventHandler(() => ({ hello: 'world' })))
```
Comment (manniL) [inline benchmarks/h3.js:7]: as @pi0 mentioned before:
> The router is fast skipped for this case. Is it still eligible to set hasRouter flag or are there any baselines a router should handle?

Comment (RafaelGSS) [reply benchmarks/h3.js:7]: Well, it should act as a router. If it transposes `http://localhost:3000/` to this handler and only this handler. It's valid.

Comment (manniL) [reply benchmarks/h3.js:7]: It "kind of" does.

While you can also opt-in for a "real router", the naive version just checks if the URL starts with the given string. So `/abc/` would match `/abc/` but also `/abc/ajsdioasdajo`. `/` would match *every* URL here.

Comment (RafaelGSS) [reply benchmarks/h3.js:7]: So, technically it shouldn't be eligible to set the `hasRouter` flag. Once it clearly has advantages in comparison to routers.

Comment (manniL) [reply benchmarks/h3.js:7]: Makes sense 👍

**Review by RafaelGSS — APPROVED** [review]

**Pull request merged**